### PR TITLE
Add azure client lib version to metadata, support flag-override of az…

### PR DESCRIPTION
**Pull request by stfeng**
…ure client lib version.

### CODE REVIEW

**Comment by stfeng** [comment]
@yuyantingzero please take a look, thanks!

File: perfkitbenchmarker/benchmarks/object_storage_service_benchmark.py [hunk 1]
```diff
@@ -72,6 +72,9 @@
 flags.DEFINE_string('boto_file_location', None,
                     'The location of the boto file.')
 
+flags.DEFINE_string('azure_lib_version', None,
```
Comment (yuyantingzero) [inline perfkitbenchmarker/benchmarks/object_storage_service_benchmark.py]: I think you can simply use a empty string as default value.

File: perfkitbenchmarker/benchmarks/object_storage_service_benchmark.py [hunk 2]
```diff
@@ -898,7 +916,10 @@ def Prepare(benchmark_spec):
 
   vms[0].Install('pip')
   vms[0].RemoteCommand('sudo pip install python-gflags==2.0')
-  vms[0].RemoteCommand('sudo pip install azure')
+  azure_version_string = ''
```
Comment (yuyantingzero) [inline perfkitbenchmarker/benchmarks/object_storage_service_benchmark.py]: If FLAGS.azure_lib_version defaults to empty string, this logic is not needed.

File: perfkitbenchmarker/benchmarks/object_storage_service_benchmark.py [hunk 3]
```diff
@@ -181,6 +184,20 @@ def _JsonStringToPercentileResults(results, json_input, metric_name,
...
 
 
+def _GetAzureClientLibVersion(vm):
+  """ This function returns the version of azure client installed on a vm
+
+  Args:
+    vm: the VM to get the azure client version from.
+
+  Returns:
+    version: The version string of the azure python client.
```
Comment (yuyantingzero) [inline perfkitbenchmarker/benchmarks/object_storage_service_benchmark.py]: No need to specify the name of return variable.
Returns:
   The version string ...

File: perfkitbenchmarker/benchmarks/object_storage_service_benchmark.py (hunk 2 above)
Comment (yuyantingzero) [inline perfkitbenchmarker/benchmarks/object_storage_service_benchmark.py]: Never mind, you will need to add '==' if we are not using latest cli.

File: perfkitbenchmarker/benchmarks/object_storage_service_benchmark.py [hunk 4]
```diff
@@ -181,6 +184,20 @@ def _JsonStringToPercentileResults(results, json_input, metric_name,
         metadata))
 
 
+def _GetAzureClientLibVersion(vm):
+  """ This function returns the version of azure client installed on a vm
```
Comment (yuyantingzero) [inline perfkitbenchmarker/benchmarks/object_storage_service_benchmark.py]: Period at the end of the sentence.

**Comment by yuyantingzero** [comment]
2 small comments, LGTM

File: perfkitbenchmarker/benchmarks/object_storage_service_benchmark.py (hunk 4 above)
Comment (stfeng) [inline perfkitbenchmarker/benchmarks/object_storage_service_benchmark.py]: Done.

File: perfkitbenchmarker/benchmarks/object_storage_service_benchmark.py (hunk 3 above)
Comment (stfeng) [inline perfkitbenchmarker/benchmarks/object_storage_service_benchmark.py]: Done.

**Comment by stfeng** [comment]
Thanks for the CR. Will submit once I finish my testing.

**Pull request merged**